apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,959 @@
+        return Promise.reject('default adapter is not existed!!');
+      }
+
+      // Save the connecting address
+      this.connectingAddress = device.address;
+      // Fire 'connecting' event.
+      var event = {
+        type: 'connecting',
+        detail: {
+          address: device.address
+        }
+      };
+      this._emitEvent(event);
+      Debug('_connect(): before start to connect, stop discovery first');
+
+      // Note on Bluedroid stack, discovery has to be stopped before connect
+      // (i.e., call stopDiscovery() before connect()) 
+      // otherwise stack callbacks with connect failure.
+      return this._defaultAdapter.stopDiscovery().then(() => {
+        Debug('_connect(): start connecting device, ' + 
+              'address = ' + device.address);
+        return this._defaultAdapter.connect(device).then(() => {
+          Debug('_connect(): resolve, already connected with address = ' + 
+                device.address);
+        }, () => {
+          // No available profiles are connected. Reset connecting address. 
+          this.connectingAddress = null;
+          // Fire 'disconnected' event.
+          event = {
+            type: 'disconnected',
+            detail: {
+              address: device.address
+            }
+          };
+          this._emitEvent(event);
+          Debug('_connect(): reject with connection failed');
+          return Promise.reject('connection failed');
+        });
+      }, () => {
+        // Cannot connect with the device since stopDiscovery failed. 
+        this.connectingAddress = null;
+        // Fire 'disconnected' event.
+        event = {
+          type: 'disconnected',
+          detail: {
+            address: device.address
+          }
+        };
+        this._emitEvent(event);
+        Debug('_connect(): reject with stop discovery failed');
+        return Promise.reject('stop discovery failed');
+      });
+    },
+
+    /**
+     * The method will disconnect the input device with the device's adapter.
+     *
+     * @access public
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothDevice} device
+     * @returns {Promise}
+     */
+    disconnect: function btcm_disconnect(device) {
+      if (!this._defaultAdapter) {
+        return Promise.reject('default adapter is not existed!!');
+      }
+
+      return this._defaultAdapter.disconnect(device).then(() => {
+        Debug('disconnect(): onsuccess(): resolve');
+      }, () => {
+        Debug('disconnect(): onerror(): reject with disconnect failed');
+        return Promise.reject('disconnect failed');
+      });
+    },
+
+    /**
+     * The method will get all connected devices profiles 
+     * which we are interested in. Profile: HFP, A2DP.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @returns {Promise} resolve: connectedDevices
+     * @returns {Promise} reject: reason
+     */
+    _getConnectedDevicesWithHFPAndA2DP: 
```

Comment:
How about `_getConnectedDevicesFromPlatform`? This avoids hard coded the profile name.